dask/tests/test_sharedict.py

```diff
@@ -0,0 +1,97 @@
+from collections import Mapping
+
+import pytest
+from toolz import merge
+
+from dask.sharedict import ShareDict, sortkey
+
+
+a = {'x': 1, 'y': 2}
+b = {'z': 3}
+c = {'w': 2}
+
+
+def test_core():
+    s = ShareDict()
+    assert isinstance(s, Mapping)
+
+    s.update(a)
+    s.update(b)
+
+    assert s['x'] == 1
+    with pytest.raises(KeyError):
+        s['abc']
+
+    with pytest.raises((NotImplementedError, TypeError)):
+        s['abc'] = 123
+
+
+def test_structure():
+    s = ShareDict()
+    s.update(a)
```

Comment:
Update remains valid.  We only use update_with_key when we explicitly provide a keyname for the mapping.